### PR TITLE
[SQL] Support for CUBE and ROLLUP

### PR DESCRIPTION
**Pull request by mihaibudiu**
Is this a user-visible change (yes/no): yes

This is a rather inefficient implementation of cube and rollup. This will have to be reworked when we solve #1341 
But we should be able to reuse some of this code at that point.
There are lots of commented-out tests because I didn't have time to enable all of them.
Note that some of the functions mentioned in the documentation are not yet fully supported (e.g., grouping).

### CODE REVIEW

**Comment by mihaibudiu** [comment]
The CI seems to fail because rustfmt panics.

**Comment by mihaibudiu** [comment]
This PR has some very small changes to the DBSP crate: fixing some comments and adding a new Tup0 type.
One has to have a tuple with zero fields. It's just mathematics.

File: docs/sql/grammar.md [hunk 1]
```diff
@@ -189,6 +191,27 @@ to `GROUP BY`.
...
+     <td><code>GROUPING(</code>expression, [ expression ]<code>)</code></td>
+     <td>Returns a bit vector of the grouping expressions</td>
+  </tr>
+  <tr>
+     <td><code>GROUP_ID()</code></td>
+     <td>Returns an integer that uniquely identifies the combination of grouping keys</td>
+  </tr>
+  <tr>
+     <td><code>GROUPING_ID(</code>expression, [ expression ]<code>)</code></td>
+     <td>Synonim for <code>GROUPING</code></td>
```
Comment (mohsinbeg) [inline docs/sql/grammar.md:211]:
```suggestion
     <td>Synonym for <code>GROUPING</code></td>
```

File: docs/sql/grammar.md [hunk 2]
```diff
@@ -189,6 +191,27 @@ to `GROUP BY`.
...
+### Grouping functions
+
+<table>
+  <tr>
+    <th>Function</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+     <td><code>GROUPING(</code>expression, [ expression ]<code>)</code></td>
+     <td>Returns a bit vector of the grouping expressions</td>
```
Comment (mohsinbeg) [inline docs/sql/grammar.md:203]: I was not able to follow this section. Maybe an example would help. For example what is a 

> bit vector of the grouping expressions

**Review by mohsinbeg — APPROVED** [review]

File: docs/sql/grammar.md (hunk 2 above)
Comment (mihaibudiu) [reply docs/sql/grammar.md:203]: these are not fully supported, I will improve the documentation when I finalize them

**Comment by mihaibudiu** [comment]
I will merge this PR because it contains a few fixes independent of the ROLLUP and lots of tests, some of which found bugs.
I will get started on fixing the bugs.

**Pull request merged**